packages/clay-drop-down/src/DropDownWithAdvancedItems.tsx

```diff
@@ -0,0 +1,247 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import {IItem} from './DropDownWithBasicItems';
+import Caption from './Caption';
+import ClayDropDown from './DropDown';
+import ClayDropDownGroup from './Group';
+import ClayForm from '@clayui/form';
+import Divider from './Divider';
+import Help from './Help';
+import React, {useState, useContext} from 'react';
+import Search from './Search';
+import warning from 'warning';
+
+type TType = 'checkbox' | 'group' | 'item' | 'radio' | 'radiogroup';
+
+interface IAdvancedItem extends Omit<IItem, 'type'> {
+	checked?: boolean;
+	items?: Array<IAdvancedItem>;
+	name?: string;
+	onChange?: Function;
+	separator?: boolean;
```

Comment:
I decided to remove the type `divider`, it felt weird to create a new object just to do it and ended up getting feedback that it looked better and easier to work with this, if we are fine with it I will do the same for` ClayDropDownWithBasicItems`.